### DISCUSSION TITLE
Some questions about self define function

### DISCUSSION THREAD

**Discussion by OriLiMu** [LazyVim/LazyVim#1843] [Q&A] [answered]
![image](https://github.com/LazyVim/LazyVim/assets/11295355/24bbc5da-4293-4bef-9db2-1bdce15f56e2)

Here is my config file I just write a simple function. 
My questions is :
1. What is the error msg mean? There is no uppercase letter in "type_hello"
2. How to source the config file when I save it?

**Comment by dpetka2001** [comment, marked as answer]
I'm not proficient in Lua, but from the error and the little that I know about Lua, if you define a function or variable starting with lowercase letter, then by convention `lua_ls` (the LSP language server for Lua) expects it to be a local function/variable. That's why you get that warning. This convention is used by `lua_ls` and I can't confidently give an answer as to how good practice that would be if you do it. To get rid of the warning, you could just put a `local` just before `function` or you should invoke code_actions by `<leader>ca` and it would provide you with options, one of which should be about something to ignore this kind of error either globally or only for this file or only for this line. I would suggest to put this not in your `init.lua` file, like you seem to be doing, but instead just create a `test.lua` file for example and then to source it, you do from inside Neovim `:luafile %`.
